exec/java-exec/src/main/java/org/apache/drill/exec/store/StoragePlugin.java

```diff
@@ -62,6 +63,14 @@
   */
   StoragePluginConfig getConfig();
 
+  /**
+   * Sets the user session.
+   * @param session The user session for the query.
+   */
+  void setUserSession(UserSession session);
```

Comment:
I don't think it is safe to set `UserSession` in this way. For the case when we have concurrent queries, `UserSession` from one user could be set, and before it is used, another user can override it...

Reply:
@vvysotskyi Thanks for the response.  I'm trying to get the `UserSession` into the constructor, but it's a little tricky with all the reflection.  Would that be better?

Reply:
One thing to note is that method gets called immediately after the plugin is constructed.

Reply:
In changes from this PR, it is set only but isn't obtained

Reply:
For this PR, I'm not going to use the UserSession for anything.... Just get it to the StoragePlugin.  Future work I'll use it for per-user credentials.

Reply:
`storages.getPlugin(schemaName, getSession());` might not create a plugin, but obtain it from the cache, so we have a single instance of the same plugin that will be attempted to hold different user sessions.

Reply:
If we pass the session to the constructor, there is no need for this setter.

Reply:
Fixed